core/src/main/java/org/mskcc/cbio/portal/servlet/QueryBuilder.java

```diff
@@ -239,9 +239,10 @@ protected void doPost(HttpServletRequest httpServletRequest,
             {
             	String extraMessage = "";
             	//extra message for the cases where property is missing (will happen often in transition period to this new versioning model):
-            	if (dbPortalVersion.equals("0"))
+            	if (dbPortalVersion.equals("0")) {
             		extraMessage = "The db.version property also not found in your portal.properties file. This new property needs to be added by the administrator.";
-                httpServletRequest.setAttribute(DB_ERROR, "Current DB Version: " + dbVersion + "<br/>" + "DB version expected by Portal: " + dbPortalVersion + "<br/>" + extraMessage);
+            	}
+                throw new DbVersionException("Current DB Version: " + dbVersion + "<br/>" + "DB version expected by Portal: " + dbPortalVersion + "<br/>" + extraMessage);
```

Comment:
I actually like the fact that all `forward...()`  logic is placed together. I think the exception should normally be thrown as an exception. If the code that handles the exception is placed there instead, it will make the code harder to read (in my opinion) since you would now suddenly be mixing up the normal workflow of the method with the exception handling logic.